mti_nma/steps/singlecell/singlecell.py

```diff
@@ -34,62 +35,102 @@ def __init__(
-        40X for tracking. Next we proceed with the single cell
-        croping that creates single cell isotropic volumes with
-        pixel size of 0.135um.
 
         Parameters
         ----------
+        cell_line_id: str
+            Name of the cell line where nuclei are going to be sampled from
+            AICS-13 = Lamin
+
         nsamples: int
             Number of cells to sample randomly (but with set seed) from dataset
-        """
+        '''
 
-        # Set reproducible random seed
+        import uuid
+        import numpy as np
         np.random.seed(666)
+        import pandas as pd
+        from tqdm import tqdm
+        from pathlib import Path
+        from aicsimageio import AICSImage, writers
 
-        # Set local directory to access raw data
-        raw_data_dir = Path('/Users/juliec/mti/raw_data')
+        if nsamples > 0:
 
-        # Load data from local directory, drawing random samples from among the cells
-        df = pd.read_csv(raw_data_dir / Path('ColonyMovie-ASCB2019.csv'), index_col=0)
-        df = df[['CellId', 'crop_raw', 'crop_seg']].sample(n=nsamples)
-        df = df.set_index('CellId')
+            self.manifest = pd.DataFrame([])
 
-        # Create directory to save data for this step in local staging
-        sc_data_dir = self.step_local_staging_dir / Path('singlecell_data')
-        sc_data_dir.mkdir(parents=True, exist_ok=True)
+            print("Loading data from LabKey...")
 
-        # Copy files to self.step_local_staging_dir and record in manifest
-        self.manifest = pd.DataFrame([])
-        for CellId in df.index:
-            shutil.copy(
-                src=raw_data_dir / Path('raw.ome.tif'),
-                dst=sc_data_dir / Path(f'{CellId}.raw.tif')
-            )
-            shutil.copy(
-                src=raw_data_dir / Path('segmentation.ome.tif'),
-                dst=sc_data_dir / Path(f'{CellId}.seg.tif')
+            df = query_data_from_labkey(CellLineId=cell_line_id) # 13 = Lamin
+
+            print(f"Number of FOVs available = {df.shape[0]}. Sampling {nsamples} FOVs now.")
+
+            df = df.sample(n=nsamples)
+
+            for fov_id in tqdm(df.index[:1]):
+
+                sx = df.PixelScaleX[fov_id]
+                sy = df.PixelScaleY[fov_id]
+                sz = df.PixelScaleZ[fov_id]
+
+                # C=-2 may not return the DNA channel for other cell lines. Need validation.
+                raw = AICSImage(df.ReadPathRaw[fov_id]).get_image_data('ZYX', S=0, T=0, C=-2)
+                seg = AICSImage(df.ReadPathSeg[fov_id]).get_image_data('ZYX', S=0, T=0, C= 0)
+
+                # Select one label from seg image at random
+                obj_label = np.random.randint(low=1, high=1+seg.max())
+
+                raw, seg = crop_object(
+                    raw = raw,
+                    seg = seg,
+                    obj_label = obj_label,
+                    isotropic = (sx,sy,sz))
```

Comment:
Is this setting all seg images to be the same dimensions?

Reply:
Based on the docstring and the code I'm pretty sure it is but want to be 100% sure I'm following

Reply:
Yes. All seg and raw images are going to have same scale in x, y, and z (scale=0.135um).